server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -958,23 +956,47 @@ private void checkForHeldServer(SortedMap<TServerInstance,TabletServerStatus> ts
       }
     }
 
-    private long balanceTablets() {
-      BalanceParamsImpl params = BalanceParamsImpl.fromThrift(tserverStatusForBalancer,
-          tserverStatus, migrationsSnapshot());
-      long wait = tabletBalancer.balance(params);
+    private long balanceTablets(boolean balanceUserTables) {
 
-      for (TabletMigration m : checkMigrationSanity(tserverStatusForBalancer.keySet(),
-          params.migrationsOut())) {
-        KeyExtent ke = KeyExtent.fromTabletId(m.getTablet());
-        if (migrations.containsKey(ke)) {
-          log.warn("balancer requested migration more than once, skipping {}", m);
+      Map<DataLevel,Set<KeyExtent>> partitionedMigrations =
+          new HashMap<>(DataLevel.values().length);
+      migrationsSnapshot().forEach(ke -> {
+        partitionedMigrations.computeIfAbsent(DataLevel.of(ke.tableId()), f -> new HashSet<>())
+            .add(ke);
+      });
+
+      BalanceParamsImpl params = null;
+      long wait = 0;
+      for (DataLevel dl : new DataLevel[] {DataLevel.ROOT, DataLevel.METADATA, DataLevel.USER}) {
```

Comment:
```suggestion
      for (DataLevel dl : DataLevel.values()) {
```

Reply:
I didn't use `values()` because I wanted a specific order. In theory, the order could change in future releases for some reason.

Reply:
Doing them in a specific order is nice, could add a comment about that.

Reply:
Another way to tackle this is to add a unit test that ensures the order of DataLevel.values() is `DataLevel.ROOT, DataLevel.METADATA, DataLevel.USER`.  Seems generally useful to work things in that order.

Reply:
Good point. My implementation doesn't account for any future values added to DataLevel.

Reply:
Implemented in 26a4939